executor/wasm-host/src/host.rs

```diff
@@ -1102,8 +1114,13 @@ pub fn casper_env_balance<S: GlobalStateReader, E: Executor>(
         .context_mut()
         .tracking_copy
         .get_total_balance(Key::URef(purse))
-        .expect("Total balance");
-    assert!(total_balance.value() <= U512::from(u64::MAX));
+        .map_err(|_| InternalHostError::TotalBalanceReadFailure)?;
+
+    // Note: This was adapted from an assertion. Not clear if still reasonable.
```

Comment:
That is definitely a reasonable assumption for casper mainnet chain. CSPR amount of `2^64-1 / 10^9` is insanely expensive